### PR TITLE
Add individual layer offset

### PR DESCRIPTION
**Pull request by DrSmugleaf**

### CODE REVIEW

**Comment by PJB3005** [comment]
Conflicts.

File: Robust.Client/GameObjects/Components/Renderable/SpriteComponent.cs [hunk 1]
```diff
@@ -936,6 +936,31 @@ public void LayerSetAutoAnimated(object layerKey, bool autoAnimated)
             LayerSetAutoAnimated(layer, autoAnimated);
         }
 
+        public void LayerSetOffset(int layer, Vector2 layerOffset)
+        {
+            if (Layers.Count <= layer)
+            {
+                Logger.ErrorS(LogCategory,
+                    "Layer with index '{0}' does not exist, cannot set auto animated! Trace:\n{1}",
```
Comment (PJB3005) [inline Robust.Client/GameObjects/Components/Renderable/SpriteComponent.cs]: > cannot set auto animated

🤔

File: Robust.Client/GameObjects/Components/Renderable/SpriteComponent.cs [hunk 2]
```diff
@@ -936,6 +936,31 @@ public void LayerSetAutoAnimated(object layerKey, bool autoAnimated)
...
+            }
+
+            Layers[layer].SetOffset(layerOffset);
+        }
+
+        public void LayerSetOffset(object layerKey, Vector2 layerOffset)
+        {
+            if (!LayerMapTryGet(layerKey, out var layer))
+            {
+                Logger.ErrorS(LogCategory, "Layer with key '{0}' does not exist, cannot set auto animated! Trace:\n{1}",
```
Comment (PJB3005) [inline Robust.Client/GameObjects/Components/Renderable/SpriteComponent.cs]: Ditto

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: Robust.Client/GameObjects/Components/Renderable/SpriteComponent.cs (hunk 1 above)
Comment (DrSmugleaf) [reply Robust.Client/GameObjects/Components/Renderable/SpriteComponent.cs]: Copy pasta strong. Fixed in https://github.com/space-wizards/RobustToolbox/pull/1583/commits/f6d393e448c353b072ef163be0a85cbd2ac62fbf

File: Robust.Client/GameObjects/Components/Renderable/SpriteComponent.cs (hunk 2 above)
Comment (DrSmugleaf) [reply Robust.Client/GameObjects/Components/Renderable/SpriteComponent.cs]: Fixed in https://github.com/space-wizards/RobustToolbox/pull/1583/commits/f6d393e448c353b072ef163be0a85cbd2ac62fbf

**Comment by Acruid** [comment]
There were a bunch of SpriteComponent changes yesterday, you need to verify this still works.

**Comment by DrSmugleaf** [comment]
> There were a bunch of SpriteComponent changes yesterday, you need to verify this still works.

It does still work yes.

image

**Pull request merged**